### ISSUE TITLE
Doesn't work on wayland

### ISSUE THREAD

**Issue by rabaimor** [Alex313031/thorium#519]
OS: Arch
DE: Hyprland

Issue: Can't launch with gtk auto darkness
If I launch it without that it crashes my whole DE within a few minutes of use.

I know your busy Alex, but please fix. Or tell me how to work around on my end.

**Comment by sachinchaudhary1808** [comment]
like can u select gtk in settings ?

**Comment by rabaimor** [comment]
Yes I can. After some time of use I can't even start the browser (log out and log in again).
When I can't start the browser I get this output until it stucks:

```
The fastest browser on Earth.

(process:2497601): Gtk-WARNING **: 14:08:10.712: Unknown key gtk-modules in /home/rabaimor/.config/gtk-4.0/settings.ini
[2497601:2497756:0121/140810.965035:ERROR:object_proxy.cc(577)] Failed to call method: org.freedesktop.DBus.Properties.Get: object_path= /org/freedesktop/portal/desktop: org.freedesktop.DBus.Error.InvalidArgs: No such interface “org.freedesktop.portal.FileChooser”
[2497601:2497756:0121/140810.965065:ERROR:select_file_dialog_linux_portal.cc(285)] Failed to read portal version property
[2497601:2497601:0121/140810.969501:ERROR:policy_logger.cc(157)] :components/enterprise/browser/controller/chrome_browser_cloud_management_controller.cc(163) Cloud management controller initialization aborted as CBCM is not enabled.

(process:2497601): Gtk-WARNING **: 14:08:11.063: No IM module matching GTK_IM_MODULE=fcitx found
```

**Comment by Alex313031** [comment]
@rabaimor In this case you should disable the auto dark mode flag, and run Thorium with the commandline flag --gtk-version=4

**Comment by rabaimor** [comment]
I already was. I tried disabling gpu as well but it did not help. 

`thorium-browser --ignore-gpu-blocklist --ozone-platform-hint=wayland --gtk-version=4 --ignore-gpu-blocklist --enable-features=TouchpadOverscrollHistoryNavigation --disable-gpu --disable-gpu-compositing`

```
The fastest browser on Earth.

(process:248657): Gtk-WARNING **: 12:59:14.678: Unknown key gtk-modules in /home/rabaimor/.config/gtk-4.0/settings.ini
[248657:248815:0122/125914.962840:ERROR:object_proxy.cc(577)] Failed to call method: org.freedesktop.DBus.Properties.Get: object_path= /org/freedesktop/portal/desktop: org.freedesktop.DBus.Error.InvalidArgs: No such interface “org.freedesktop.portal.FileChooser”
[248657:248815:0122/125914.962874:ERROR:select_file_dialog_linux_portal.cc(285)] Failed to read portal version property
[248657:248657:0122/125914.970916:ERROR:policy_logger.cc(157)] :components/enterprise/browser/controller/chrome_browser_cloud_management_controller.cc(163) Cloud management controller initialization aborted as CBCM is not enabled.

(process:248657): Gtk-WARNING **: 12:59:15.066: No IM module matching GTK_IM_MODULE=fcitx found

```
Also I tried vanilla chromium as well, but that did not start at all. But Thorium starts after I log in, but charshes shortly, and after that if I try to restart it cant with the output above.

**Comment by Alex313031** [comment]
@rabaimor I'm sorry but I can't really dedicate time to this, and I am doubtful I would be able to fix it even if I did. I love the amount of choice that linux brings, but it also causes stuff like this. Where everyone has a different configuration and it's hard to make large applications work seamlessly. I hate to be that guy but my suggestion is to use X11 or a more traditional distro that is actively tested by the Chromium devs like Ubuntu, Debian, or Gentoo.

**Comment by opticblu** [comment]
It works perfectly for me on Arch Hyprland and has for a year

Pretty much every sensible “doesn’t generally break everything” flag works for me, Vulkan, Skia, zero copy, GTK auto dark

This is either a hardware issue or a skill issue/Hyprland config issue

Anecdotal, but Thorium and other highly-optimized/clanged-ltoed software seem to be really sensitive to unstable overclocks, both RAM and CPU, but if it works on KDE, it’s extremely likely that it’s just a config issue and you have bad environment variables for your config somewhere 

Not trolling or claiming any one browser is the holy grail, this is my genuine experience

**Comment by hash4q** [comment]
this fixed it for me "--enable-features=UseOzonePlatform --ozone-platform=wayland" .

**Comment by gz83** [comment]
Close stale and inactive issues.
